presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -36,7 +37,10 @@
     @Override
     protected void setup(Binder binder)
     {
-        binder.bind(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
+        OptionalBinder.newOptionalBinder(binder, ThriftMetastoreClientFactory.class)
+                .setDefault()
+                .to(DefaultThriftMetastoreClientFactory.class)
+                .in(Scopes.SINGLETON);
```

Comment:
```suggestion
                .setDefault().to(DefaultThriftMetastoreClientFactory.class).in(Scopes.SINGLETON);
```

Reply:
Hmmm - I read many times to either break on all dots. Or not at all. I will leave it :)

Reply:
That’s not the rule. You break where it makes logical sense.

Reply:
updated